### PR TITLE
Use assertj fluent style in flowable-engine module.

### PR DESCRIPTION
**Pull request by dbmalkovsky**
In the `org.flowable.examples` package there were 4 files that had a mixture of unit testing styles within the same file.    They now all are using assertj tests.

### CODE REVIEW

File: modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java [hunk 1]
```diff
@@ -105,31 +105,32 @@ public void testIdentityLinks() throws Exception {
...
             latestProcessDef = repositoryService.createProcessDefinitionQuery().processDefinitionKey("process2").singleResult();
-            assertNotNull(latestProcessDef);
+            assertThat(latestProcessDef).isNotNull();
             links = repositoryService.getIdentityLinksForProcessDefinition(latestProcessDef.getId());
-            assertEquals(2, links.size());
-            assertTrue(containsUserOrGroup("user1", null, links));
-            assertTrue(containsUserOrGroup("user2", null, links));
+            assertThat(links).hasSize(2);
+            assertThat(containsUserOrGroup("user1", null, links)).isTrue();
+            assertThat(containsUserOrGroup("user2", null, links)).isTrue();
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java:117]:
```suggestion
            assertThat(links)
                    .extracting(IdentityLink::getUserId)
                    .containsExactlyInAnyOrder("user1", "user2");
```

File: modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java [hunk 2]
```diff
@@ -105,31 +105,32 @@ public void testIdentityLinks() throws Exception {
...
-            assertEquals(4, links.size());
-            assertTrue(containsUserOrGroup("userInGroup2", null, links));
-            assertTrue(containsUserOrGroup(null, "group1", links));
-            assertTrue(containsUserOrGroup(null, "group2", links));
-            assertTrue(containsUserOrGroup(null, "group3", links));
+            assertThat(links).hasSize(4);
+            assertThat(containsUserOrGroup("userInGroup2", null, links)).isTrue();
+            assertThat(containsUserOrGroup(null, "group1", links)).isTrue();
+            assertThat(containsUserOrGroup(null, "group2", links)).isTrue();
+            assertThat(containsUserOrGroup(null, "group3", links)).isTrue();
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java:133]:
```suggestion
            assertThat(links)
                    .extracting(IdentityLink::getUserId, IdentityLink::getGroupId)
                    .containsExactlyInAnyOrder(
                        tuple("userInGroup2", null),
                        tuple(null, "group1"),
                        tuple(null, "group2"),
                        tuple(null, "group3")
                    );
```

File: modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java [hunk 3]
```diff
@@ -143,34 +144,37 @@ public void testAddAndRemoveIdentityLinks() throws Exception {
...
+                    .containsExactly("group1");
 
             repositoryService.addCandidateStarterUser(latestProcessDef.getId(), "user1");
             links = repositoryService.getIdentityLinksForProcessDefinition(latestProcessDef.getId());
-            assertEquals(2, links.size());
-            assertTrue(containsUserOrGroup(null, "group1", links));
-            assertTrue(containsUserOrGroup("user1", null, links));
+            assertThat(links).hasSize(2);
+            assertThat(containsUserOrGroup(null, "group1", links)).isTrue();
+            assertThat(containsUserOrGroup("user1", null, links)).isTrue();
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java:163]:
```suggestion
            assertThat(links)
                    .extracting(IdentityLink::getUserId, IdentityLink::getGroupId)
                    .containsExactlyInAnyOrder(
                        tuple("user1", null),
                        tuple(null, "group1")
                    );
```

File: modules/flowable-engine/src/test/java/org/flowable/examples/variables/VariablesTest.java [hunk 4]
```diff
@@ -197,75 +206,77 @@ public void testLocalizeVariables() {
...
         List<String> variableNames = new ArrayList<>();
         variableNames.add("stringVar");
 
         // getVariablesInstances via names
         variableInstances = runtimeService.getVariableInstances(processInstance.getId(), variableNames);
-        assertEquals(1, variableInstances.size());
-        assertEquals("stringVar", variableInstances.get("stringVar").getName());
-        assertEquals("coca-cola", variableInstances.get("stringVar").getValue());
+        assertThat(variableInstances.get("stringVar"))
+                .extracting("name", "value")
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/examples/variables/VariablesTest.java:219]: Let's not use reflection extraction of field names. Why not use `VariableInstance::getName, VariableInstance::getValue` instead? There are other places where this is done as well.

File: modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java (hunk 1 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java:117]: done

File: modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java (hunk 2 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java:133]: done

File: modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java (hunk 3 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/examples/bpmn/authorization/StartAuthorizationTest.java:163]: done

File: modules/flowable-engine/src/test/java/org/flowable/examples/variables/VariablesTest.java (hunk 4 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/examples/variables/VariablesTest.java]: Done

**Comment by dbmalkovsky** [comment]
I undid the long list of assertions as suggested.   I had gone back and forth and whether to do that or not; obviously made the wrong choice :)

**Comment by filiphr** [comment]
Thanks a lot for adapting this @dbmalkovsky.

**Pull request merged**